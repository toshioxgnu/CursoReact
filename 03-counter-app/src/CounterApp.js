import React, {useState} from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState(value);

    //handleAdd
    const handdleAdd = () => {
        setCounter( counter + 1 );
    }

    //handleReset
    const handleReset = () => {
        setCounter( value )
    }

    const handleSustract = () => {
        setCounter( counter -1 )
    }

    return (
        <>
            <h1> Counter App </h1>
            <h2> { counter } </h2>

            <button onClick={ handdleAdd } >+1</button>
            <button onClick={ handleReset } >Reset</button>
            <button onClick={ handleSustract } >-1</button>
            
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;