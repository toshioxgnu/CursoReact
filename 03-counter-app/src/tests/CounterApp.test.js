import CounterApp from "../CounterApp";
import { shallow } from 'enzyme';


describe(' Pruebas en Counter App ', () => {

    let  wrapper  = shallow( <CounterApp value={ 10 } /> );
    
    beforeEach( () => {
        wrapper  = shallow( <CounterApp value={ 10 } /> );
    });
    
    test('debe mostrar counter app correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar valor por defecto de 100', () => {
        const counterText = wrapper.find('h2').text();
        expect( counterText ).toBe( ' 10 ' );
    });

    test('Debe incrementar con el boton de +1 ', () => {
        const btn1 = wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('11');

    });

    test('Debe restar con el boton de -1 ', () => {
        const btn1 = wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('9');

    });

    test('Debe colocar el valor por defecto con el boton reset ', () => {
        const wrapper  = shallow( <CounterApp value={ 10 } /> );
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('10');
    })
    
});