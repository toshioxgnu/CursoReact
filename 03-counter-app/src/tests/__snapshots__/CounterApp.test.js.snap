// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en Counter App  debe mostrar counter app correctamente  1`] = `
<Fragment>
  <h1>
     Counter App 
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
